doc/User_Management/User_Management.md

```diff
@@ -0,0 +1,344 @@
+# Declarative Local User Management
+
+## Revision
+
+| Rev | Date           | Author      | Change Description                                                                                                                                   |
+|:---:|:--------------:|:-----------:|:----------------------------------------------------------------------------------------------------------------------------------------------------:|
+| 1.0 | June 16, 2025  | Manoharan Sundaramoorthy   | Initial HLD                                                                                                                                          |
+
+
+## 1. Scope
+This document describes the high-level design for a new feature that provides a declarative and persistent method for managing local user accounts on a SONiC device. This feature allows administrators to define local users, including their roles, password hashes, SSH keys, and **security policies** including **login attempt limits**.
+
+This ensures that local user accounts are consistently applied with robust security postures and persist across reboots and upgrades.
+
+## 2. Definitions/Abbreviations
+
+| Abbreviation | Definition                                                                |
+|:------------:|:--------------------------------------------------------------------------|
+| **`userd`** | The new User Daemon proposed in this design.                              |
+| **PAM** | Pluggable Authentication Modules                                          |
+
+
+## 3. Overview
+This feature introduces a new dedicated daemon, **`userd`**, which manages the full lifecycle of local users based on definitions in `CONFIG_DB`. It simplifies management by providing a user-friendly CLI, mapping abstract roles (`administrator`, `operator`) to specific Linux groups, and enforcing security policies. This provides a solution for managing secure, persistent local user accounts.
+
+## 4. Requirements
+### 4.1 Functional Requirements
+1.  The system must allow an administrator to define a local user account declaratively.
+2.  The user definition must support:
+    * Username and a pre-hashed password.
+    * A role, limited to either **`administrator`** or **`operator`**.
+    * Authorized SSH keys (statically defined).
+3.  The system will **auto-generate** a unique UID for each new user.
```

Comment:
Will adding a removing the same user end up using the same UID (e.g. some kind of hash?)? Will the same user have the same UID across multiple SONiC installations, or will it be sequential?